adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java

```diff
@@ -217,26 +304,78 @@ public static Requirement getRequirement(HttpServletRequest httpServletRequest,
     {
         Requirement aResource = null;
         
-        // Start of user code getRequirement
-        aResource = requirements.get(requirementId);
+        // Start of user code getRequirement_storeInit
+        // End of user code
+        Store store = storePool.getStore();
+        URI uri = RMToolResourcesFactory.constructURIForRequirement(requirementId);
+        try {
+            aResource = store.getResource(storePool.getDefaultNamedGraphUri(), uri, Requirement.class);
+        } catch (NoSuchElementException e) {
+            log.error("Resource: '" + uri + "' not found");
+            throw new WebApplicationException("Failed to get resource: '" + uri + "'", e, Status.NOT_FOUND);
+        } catch (StoreAccessException | ModelUnmarshallingException  e) {
+            log.error("Failed to get resource: '" + uri + "'", e);
+            throw new WebApplicationException("Failed to get resource: '" + uri + "'", e, Status.INTERNAL_SERVER_ERROR);
+        } finally {
+            storePool.releaseStore(store);
+        }
+        // Start of user code getRequirement_storeFinalize
         // End of user code
+        
+        // Start of user code getRequirement
+       // End of user code
         return aResource;
     }
 
     public static Boolean deleteRequirement(HttpServletRequest httpServletRequest, final String requirementId)
     {
         Boolean deleted = false;
+        // Start of user code deleteRequirement_storeInit
+        // End of user code
+        Store store = storePool.getStore();
+        URI uri = RMToolResourcesFactory.constructURIForRequirement(requirementId);
+        if (!store.resourceExists(storePool.getDefaultNamedGraphUri(), uri)) {
+            log.error("Cannot delete a resource that does not already exists: '" + uri + "'");
```

Comment:
I think you can just fire the delete query directly. DELETE shall be idempotent and Store#delete is also idempotent as per SPARQL spec.

Reply:
Sure. Still tempted to throw an exception  for some reason.

Reply:
I think a cooler way would be to keep a log of deleted URIs and return 410 Gone for them :)

Reply:
In the future :-)